### PR TITLE
driver-adapters: libsql/turso adapter

### PR DESCRIPTION
**Pull request by aqrln**

### CODE REVIEW

File: query-engine/driver-adapters/src/conversion.rs
Comment (aqrln) [inline query-engine/driver-adapters/src/conversion.rs:37]: TODO: test `Bytes` support in other adapters.

For libSQL it's essential that we pass these as `ArrayBuffer`s as we don't have sufficient type information after this point in general case.

If any other adapters require base64-encoded strings, either they will need to communicate their preference similar to how they report the `flavour`, or additional mapping needs to happen on JS side.

File: query-engine/driver-adapters/src/proxy.rs
Comment (aqrln) [inline query-engine/driver-adapters/src/proxy.rs:260]: TODO: this is sufficient for now to make smoke tests work but quaint supports more formats, do we need those?

File: query-engine/driver-adapters/src/proxy.rs
Comment (aqrln) [inline query-engine/driver-adapters/src/proxy.rs:278]: TODO for future: unify these two branches, get rid of JSON arrays here and use base64-encoded strings everywhere (in results of the queries, not in parameters)

File: query-engine/driver-adapters/src/proxy.rs
Comment (aqrln) [reply query-engine/driver-adapters/src/proxy.rs:179]: Also `Boolean` needs more cases, just got `thread 'tokio-runtime-worker' panicked at 'Expected a boolean in column boolean_column, found "1"', query-engine/driver-adapters/src/proxy.rs:228:25`

File: query-engine/driver-adapters/js/smoke-test-js/package.json
Comment (janpio) [inline query-engine/driver-adapters/js/smoke-test-js/package.json:14]: Migrations are probably a problem for the remote ones (including embedded replica)?

File: query-engine/driver-adapters/src/queryable.rs
Comment (aqrln) [inline query-engine/driver-adapters/src/queryable.rs:120]: This should be handled by the adapters themselves in the future

File: query-engine/driver-adapters/js/smoke-test-js/package.json
Comment (aqrln) [reply query-engine/driver-adapters/js/smoke-test-js/package.json:14]: We'll have a way to run migrations with driver adapters, most likely using a design similar to the test setup, with migrate sending queries to the external node process via JSON-RPC. Initially just to make testing against sqld/turso possible but the user-facing functionality may be based on that as well. We discussed this with Miguel briefly today.

At this moment though, running proper tests is only possible with local file, and smoke tests require manual `turso db shell DB < migration.sql && turso db shell DB < insert.sql` when using a remote DB. I can add a separate npm script for that.

File: query-engine/driver-adapters/src/proxy.rs
Comment (jkomyno) [reply query-engine/driver-adapters/src/proxy.rs:179]: > Also `Boolean` needs more cases:
The original idea was to keep the implementation simple in Rust, and adapt the JS adapters (pun not intended) to it on the TypeScript side. So, if e.g. Turso represents booleans with `0` and `1`, they should be respectively transformed into `false` and `true` in TypeScript

File: query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs
Comment (aqrln) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs:347]: remove before merging

Comment (jkomyno) [inline query-engine/connector-test-kit-rs/query-engine-tests/tests/new/occ.rs:347]:
```suggestion
```

**Review by miguelff — APPROVED** [review]
Let's ship this an iterate @aqrln!

**Comment by aqrln** [comment]
Merging as per discussion with @miguelff.

Next steps to do in follow-up PRs:
1. Move the conversion logic to the appropriate layer as requested by @jkomyno.
2. Abstract away common JS conversions for driver adapters (bigint->string, bytes etc) as a reusable piece and move it to driver-adapter-utils.
3. Concurrency control.
4. Investigate upsert tests.
5. Skip tests failing due to mismatching error codes, pending implementation of error parsing.

**Review by aqrln — COMMENTED** [review]
Some unrelated tests for normal QE are failing since 7de9f9b04d1de227dcd32a35c67b56ac50fe720f, blocking the PR from being merged for now.

**Pull request merged**
